src/Packagist/WebBundle/Controller/WebController.php

```diff
@@ -38,16 +39,17 @@ public function indexAction()
 
     public function searchFormAction(Request $req)
     {
-        $form = $this->createForm(new SearchQueryType, new SearchQuery);
+        $form = $this->createForm(SearchQueryType::class, new SearchQuery(), [
+            'action' => $this->generateUrl('search.ajax'),
+            'method' => 'GET',
+        ]);
 
         $filteredOrderBys = $this->getFilteredOrderedBys($req);
         $normalizedOrderBys = $this->getNormalizedOrderBys($filteredOrderBys);
 
         $this->computeSearchQuery($req, $filteredOrderBys);
 
-        if ($req->query->has('search_query')) {
-            $form->submit($req);
-        }
+        $form->handleRequest($req);
```

Comment:
The problem AFAIK is that form->handleRequest isn't very smart and it just accepts the form as submitted and valid even if there is nothing submitted in this case, maybe because our whole form is optional, not sure, but it's not really a big deal to use the form->submit() manually IMO, we don't have to rely on the magic everywhere. It's not deprecated or anything AFAIK.